opendatahub-io/odh-dashboard
# [RHOAIENG-12310] Error cases for storage class admin table

https://issues.redhat.com/browse/RHOAIENG-12310

## Description
Show error/warning alerts for scenarios where individual storage class config values or the entire config are malformed in some way. Introduce abilities to correct those mistakes possibly made via the Openshift UI within the ODH dashboard's storage class table.

<img width="1434" alt="image" src="https://github.com/user-attachments/assets/7ef334e0-1ca8-4990-8f26-aba64ed04a4c">

**Unreadable config demo:**
https://github.com/user-attachments/assets/86b3cda6-f5c4-41cf-bcdc-92f8787f9b61

**Invalid data, default, enable config values:**
https://github.com/user-attachments/assets/35a9d29c-f889-4101-a2f2-690fe4a8afbb

**Invalid Display name demo:**
https://github.com/user-attachments/assets/af5d969f-450c-4303-a1e5-3057554d114d

**Invalid Description demo:**
https://github.com/user-attachments/assets/8f4b14a3-b97f-4035-b24c-0f8d28a885b7

**Invalid Display name and description demo:**
https://github.com/user-attachments/assets/9fd6f80f-5590-42e9-b524-3c7db61c7b16

(cc @xianli123)

## How Has This Been Tested?
1. Edit storage classes within in Openshift via `/search/ns/opendatahub?q=&kind=storage.k8s.io~v1~StorageClass` by clicking on individual row actions, `Edit annotations`, then update the `opendatahub.io/sc-config` value to force invalid data.

With the feature flag enabled, visit the storage class table's page; `/storageClasses?devFeatureFlags=disableStorageClasses`

2. Verify the cases outlined in the JIRA issue are covered and match the referenced designs.
3. Verify all storage class edit modals are updated according to how/where they were opened from.
4. Verify once updating a given row or individual cell that the data is reset and no longer considered "Corrupted".

## Test Impact
Added cypress test covering multiple error cases

## Request review criteria:
Self checklist (all need to be checked):

If you have UI changes: 

After the PR is posted & before it merges:
